From 49f90a69ecd6dbf5d021a08e6df88a94b967e81b Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Date: Tue, 9 Jan 2018 11:37:59 +0100
Subject: [PATCH 0353/1051] octeontx: move domain creation flags for TIM & DPI

Move TIM and DPI domain creation flags to correct location.

Signed-off-by: Lukasz Bartosik <lukasz.bartosik@cavium.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
index 23d66a04e14a..a7403e0f683a 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/octeontx_main.c
@@ -999,8 +999,8 @@ int octeontx_create_domain(const char *name, int type, int sso_count,
 			dev_err(octtx_device, "Failed to create TIM domain\n");
 			goto error;
 		}
+		domain->tim_domain_created = true;
 	}
-	domain->tim_domain_created = true;
 
 	domain->dpi_vf_count = dpi_count;
 	if (domain->dpi_vf_count > 0) {
@@ -1012,8 +1012,8 @@ int octeontx_create_domain(const char *name, int type, int sso_count,
 			dev_err(octtx_device, "Failed to create DPI domain\n");
 			goto error;
 		}
+		domain->dpi_domain_created = true;
 	}
-	domain->dpi_domain_created = true;
 
 	domain->zip_vf_count = zip_count;
 	if (domain->zip_vf_count) {
-- 
2.17.1

